test/unit/tracks/tracks.test.js

```diff
@@ -342,3 +342,28 @@ if (Html5.supportsNativeTextTracks()) {
     emulatedTt.on('addtrack', addtrack);
   });
 }
+
+test('should check for text track changes when using non native text tracks', function() {
+  let origFeaturesNativeTextTracks = Tech.prototype['featuresNativeTextTracks'];
+  Tech.prototype['featuresNativeTextTracks'] = true;
+  let Player = Component.getComponent('Player');
+  let origTextTracks = Player.prototype.textTracks;
+  let numTextTracksCalls = 0;
+  Player.prototype.textTracks = function() {
```

Comment:
Yeah, the tricky part about testing it was that the testing needed to be done during player setup, making it hard to listen for specific events since they could’ve happened by the time we had a player reference. That said, since this is a unit test, it is better suited to simply check that emulateTextTracks on Tech fires `texttrackchange` before listening (as you mentioned), and we can avoid the entire player setup and nativeTextTracks setting.